unlock-js/src/unlockService.js

```diff
@@ -56,11 +71,11 @@ export default class UnlockService extends EventEmitter {
       throw new Error(Errors.MISSING_WEB3)
     }
 
-    let version = this.versionForAddress[address]
-    if (!version) {
+    let version = this.versionForAddress[address.toLowerCase()]
```

Comment:
Does the address need to be lower case here? What occurs when it isn't?

Reply:
there are 2 versions of addresses used in the ethereum world, the all-lower-cased address, and the checksummed address, which has a mixture of upper and lower case letters. If we don't lower-case the address, we will end up constructing 2 copies of the contract and memoizing both, which uses quite a bit of memory unnecessarily (about 3k per contract)

Reply:
actually, this change is not needed in the web3 side of things, I'll remove it from there, and we can apply your suggestions below